reference/extensions/deployers.rst

```diff
@@ -1,5 +1,76 @@
 .. _reference_extensions_deployers:
 
 Deployers
----------
+=========
 
+Deployers are a mechanism to facilitate copying files form one folder, usually the Conan cache, to user folders.
+While Conan provides two built-in ones (``full_deploy`` and ``direct_deploy``), users can easily manage their own
+with ``conan config install``.
+
+Deployers run before generators, and they can change the target folders.
+For example, if the ``--deploy=full_deploy`` deployer runs before ``CMakeDeps``,
+the files generated by ``CMakeDeps`` will point to the local copy in the user folder done by the ``full_deploy`` deployer,
+and not to the Conan cache. Multiple deployers can be specified by supplying more than one ``--deploy=`` argument,
+and they will be ran in order of appearance.
+
+Deployers can be multi-configuration. Running ``conan install . --deploy=full_deploy`` repeatedly for different profiles
+can achieve a fully self-contained project, including all the artifacts, binaries, and build files.
+This project will be completely independent of Conan and no longer require it at all to build.
+
+
+Built-in deployers
+------------------
+
+.. _reference_extensions_deployer_full_deploy:
+
+full_deploy()
+^^^^^^^^^^^^^
+
+Deploys each package folder of every dependency to your recipe's ``output_folder`` in a subfolder tree based on:
+
+#. The build context
+#. The dependency name and version
+#. The build type
+#. The build arch
+
+Such that every dependency will end up in a folder such as:
+
+``[OUTPUT_FOLDER]/host/dep/0.1/Release/x86_64``
+
+
+.. _reference_extensions_deployer_direct_deploy:
+
+direct_deploy()
+^^^^^^^^^^^^^^^
+
+Same as ``full_deploy()```, but only processes your recipe's *direct* dependencies.
```

Comment:
I think that we should add a little piece of code for both functions. IMO, I think it's not clear enough how to use those built-in functions. What do you think?

Reply:
Apart from that, are there any args/kwargs that the users should be aware of?

Reply:
I think we should in fact not let the user know that they are functions, only names that they can call, which saves us from having to deal with these concerns. Does that sound good to you?

Reply:
Sure! 😉